### PR TITLE
ROBJSTORE-125 Move MongoClient under SyncUser

### PR DESCRIPTION
**Pull request by jbreams**
This moves the factory for MongoClient to SyncUser from App so that mongo operations are performed by a specific user rather than whatever the current user of the App is. It also renamed RemoteMongoClient/DatabaseCollection to just MongoClient/Database/Collection.

### CODE REVIEW

**Comment by jbreams** [comment]
A few things:

1) I didn't realize you don't use clang-format in this repo before running `git clang-format -f`. I use the clang-format config from realm-core so at least it's consistent with that? If you don't like the formatting changes I can go back and fix them. Sorry for the noise.
2) I'll rename the remote_mongo_* files to just mongo_* files if you approve the substantive changes I've made. I didn't want to create more noise/confusion during review by renaming them and making a bunch of changes at the same time.

**Comment by nirinchev** [comment]
This will conflict so nicely with https://github.com/realm/realm-object-store/pull/1113 😄 That one was a fairly trivial refactor though, so I think it'd be best if I just redid it after you merge your PR.

File: src/sync/remote_mongo_collection.cpp
Comment (jsflax) [inline src/sync/remote_mongo_collection.cpp:94]: Newline between params

File: src/sync/remote_mongo_collection.cpp
Comment (jsflax) [inline src/sync/remote_mongo_collection.cpp:183]: Why the heavy indentation?

File: src/sync/remote_mongo_collection.hpp
Comment (jsflax) [inline src/sync/remote_mongo_collection.hpp:24]: Newline before angled imports (I know it wasn't there before, but it should have been)

File: src/sync/remote_mongo_collection.hpp
Comment (jsflax) [inline src/sync/remote_mongo_collection.hpp:326]: In general our constructor inits has been on the same indentation as the constructor declaration.

File: src/sync/sync_user.hpp
Comment (jsflax) [inline src/sync/sync_user.hpp:238]: I'm not sure any of the SDKs actually use `SyncUserContext` anymore. @nirinchev does .NET?

**Review by jsflax — COMMENTED** [review]
Nice job 👍  mostly some style remarks and a question around whether or not SyncUserContext is still used.

File: src/sync/remote_mongo_collection.hpp
Comment (jbreams) [reply src/sync/remote_mongo_collection.hpp:24]: Done

**Review by jbreams — COMMENTED** [review]
I've undone most of the clang-format changes so this is less noisy.

**Review by jsflax — APPROVED** [review]

**Review by nirinchev — APPROVED** [review]

**Pull request merged**
